src/srcset.js

```diff
@@ -180,119 +165,110 @@ export class Srcset {
+    } else {
+      index = this.selectByDpr_(dpr);
     }
-    return this.getLast();
+    return this.sources_[index].url;
   }
 
   /**
    * @param {number} width
-   * @param {number} dpr
    * @return {number}
    * @private
    */
-  selectByWidth_(width, dpr) {
-    const length = this.sources_.length;
-    let prevWidth = -Infinity;
-    for (let i = length - 1; i >= 0; i--) {
-      const source = this.sources_[i];
-      const sourceWidth = source.width / dpr;
-      // First candidate width that's equal or higher than the requested width
-      // will stop the search.
-      if (sourceWidth >= width) {
-        // The right value is now between `i` and `i + 1` - select the one
-        // that is closer with a slight preference toward higher numbers.
-        const delta = sourceWidth - width;
-        const prevDelta = (width - prevWidth) * 1.1;
-        // If smaller size is closer, enfore minimum ratio between
-        // requested width and prevWidth to ensure image isn't too distorted.
-        if (prevDelta < delta && width / prevWidth <= 1.2) {
-          return i + 1;
-        }
-        return i;
+  selectByWidth_(width) {
+    const sources = this.sources_;
+    let minIndex = 0;
+    let minScore = Infinity;
+    let minWidth = Infinity;
+
+    for (let i = 0; i < sources.length; i++) {
+      const sWidth = sources[i].width;
+      const score = Math.abs(sWidth - width);
+
+      // Select the one that is closer with a slight preference toward larger
+      // widths. If smaller size is closer, enforce minimum ratio to ensure
+      // image isn't too distorted.
+      if (score <= minScore * 1.1 || width / minWidth > 1.2) {
+        minIndex = i;
+        minScore = score;
+        minWidth = sWidth;
+      } else {
+        break;
       }
-      prevWidth = sourceWidth;
     }
-    // Use the first (maximum) value.
-    return 0;
+    return minIndex;
   }
 
   /**
-   * @param {number} _width
    * @param {number} dpr
    * @return {number}
    * @private
    */
-  selectByDpr_(_width, dpr) {
-    let minIndex = -1;
-    let minScore = 1000000;
-    for (let i = 0; i < this.sources_.length; i++) {
-      const source = this.sources_[i];
-      // Default DPR = 1.
-      const sourceDpr = source.dpr || 1;
-      const score = Math.abs(sourceDpr - dpr);
-      if (score < minScore) {
-        minScore = score;
+  selectByDpr_(dpr) {
+    const sources = this.sources_;
+    let minIndex = 0;
+    let minScore = Infinity;
+
+    for (let i = 0; i < sources.length; i++) {
+      const score = Math.abs(sources[i].dpr - dpr);
+      if (score <= minScore) {
         minIndex = i;
+        minScore = score;
+      } else {
+        break;
       }
     }
     return minIndex;
   }
 
   /**
-   * Returns the last source in the srcset, which is the default source.
-   * @return {!SrcsetSourceDef}
+   * Returns all URLs in the srcset.
+   * @return {!Array<string>}
    */
-  getLast() {
```

Comment:
Welp, there's a test in `test-amp-image-lightbox.js` that uses this and needs to be updated too.

Reply:
Done.